### ISSUE TITLE
XamlTypeInfo forgets to generate type information if only used as type

### ISSUE THREAD

**Issue by huoyaoyuan** [microsoft/microsoft-ui-xaml#6232]
### Describe the bug

XamlTypeInfo forgets to generate type information if the type is only used as type xaml, not object or property.

### Steps to reproduce the bug

1. Create a project with WinUI3, and install CommunityToolkit.WinUI.UI 7.1.1-preview3
2. Create an enum named `MyEnum`
3. Add following xaml:
```xaml
    xmlns:wctk="using:CommunityToolkit.WinUI.UI"
```

```xaml
<ListView ItemsSource="{wctk:EnumValues Type=local:MyEnum}" />
```
4. The markup extension receives a null type, and fails to provide values of the enum.

### Expected behavior

Enum values shown in the ListView.

### Screenshots

N/A

### NuGet package version

WinUI 3 - Windows App SDK 1.0.0-preview3

### Windows app type

- [ ] UWP
- [X] Win32

### Device form factor

Desktop

### Windows version

May 2021 Update (19043)

### Additional context

The workaround is to create a dummy object:
```xaml
        <StackPanel.Resources>
            <local:MyEnum x:Key="__Dummy">0</local:MyEnum>
        </StackPanel.Resources>
```

Having a control with the enum as a property will make it generates correctly.

**Comment by StephenLPeters** [comment]
@RealTommyKlein and @evelynwu-msft FYI

**Comment by RealTommyKlein** [comment]
Thanks for the report - this will be fixed in Windows App SDK 1.1.  As an alternative workaround, you can also add the `Microsoft.UI.Xaml.Data.Bindable` attribute to the types being assigned to the property, e.g.:
```
[Microsoft.UI.Xaml.Data.Bindable]
enum MyEnum
{
    ....
}
```

**Comment by RealTommyKlein** [comment]
This is fixed for the upcoming Windows App SDK 1.1 release - also, closing this issue as a dupe of #4161
